kv/memberlist/http_status_handler.go

```diff
@@ -0,0 +1,217 @@
+package memberlist
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/memberlist"
+)
+
+// HTTPStatusHandler is a http.Handler with status information about memberlist.
+type HTTPStatusHandler struct {
+	kvs *KVInitService
+	tpl *template.Template
+}
+
+// StatusPageData represents the data passed to the template rendered by HTTPStatusHandler
+type StatusPageData struct {
+	Now              time.Time
+	Memberlist       *memberlist.Memberlist
+	SortedMembers    []*memberlist.Node
+	Store            map[string]ValueDesc
+	SentMessages     []Message
+	ReceivedMessages []Message
+}
+
+// NewHTTPStatusHandler creates a new HTTPStatusHandler that will render the provided template using the data from StatusPageData.
+func NewHTTPStatusHandler(kvs *KVInitService, tpl *template.Template) HTTPStatusHandler {
+	return HTTPStatusHandler{kvs, tpl}
+}
+
+func (h HTTPStatusHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
```

Comment:
Note to reviewers: this function is the same exact code of the previous `KVInitService.ServeHTTP()`

Reply:
Yes, it's actually straightforward if you check individual commits, I did the code extraction in the first commit.